pages/api/reviews/index.js

```diff
@@ -0,0 +1,59 @@
+import dbConnect from "@/db/connect";
+import Review from "@/db/models/Review";
+import Activity from "@/db/models/Activity";
+import User from "@/db/models/User";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../auth/[...nextauth]";
+import { getToken } from "next-auth/jwt";
+
+export default async function handler(request, response) {
+  await dbConnect();
+  const token = await getToken({ req: request });
+  const session = await getServerSession(request, response, authOptions);
+  const userId = token?.sub;
+  const userName = token?.name;
+  const picture = token?.picture;
+
+  if (!session) {
+    return response.status(401).json({ message: "Nicht autorisiert" });
+  }
+
+  if (request.method === "POST") {
+    try {
+      const { activityId, rating } = request.body;
+      if (!activityId || !rating) {
+        return response
+          .status(400)
+          .json({ message: "Aktivitäts-ID und Bewertung sind erforderlich" });
+      }
+
+      let user = await User.findOne({ userId });
```

Comment:
```suggestion
      const user = await User.findOne({ userId });
```

Reply:
In this case, the code would not work if we use const instead of let, if User.findOne returns undefined or null, we would not be able to reassign the value of the  new created user ....